### ISSUE TITLE
Unable to create CurvedSegment between two sketches

### ISSUE THREAD

**Issue by AIRCAP** [chbergmann/CurvedShapesWorkbench#49]
The sketches have the same number of points and edges - so the makeRibsSameShape() code branch is triggered - however the sketches are incompatible, where one has a convex circular sector the other has a concave (which have fixed directionality and inverse order of points on the edge)

this could probably be solved with the "mareRibsInterpolate" code branch but that is not reached with the existing checks in place
![incompatible_sketches](https://github.com/user-attachments/assets/74745dfc-8c64-4cdc-bbb7-6ddc5a5808d8)

```
18:00:53  Traceback (most recent call last):
  File "~/.local/share/FreeCAD/Mod/CurvedShapes/./CurvedSegment.py", line 99, in execute
    self.makeRibs(fp)
  File "~/.local/share/FreeCAD/Mod/CurvedShapes/./CurvedSegment.py", line 129, in makeRibs
    ribs = makeRibsSameShape(fp, fp.Items, False, makeStartEnd)
  File "~/.local/share/FreeCAD/Mod/CurvedShapes/./CurvedSegment.py", line 231, in makeRibsSameShape
    newpoles.append(vectorMiddlePlane(poles1[p], poles2[p], fraction, plane))
<class 'IndexError'>: list index out of range
18:00:53  Recompute failed!
```

[incompatible_sketches.zip](https://github.com/user-attachments/files/17246921/incompatible_sketches.zip)

**Comment by AIRCAP** [comment]
there are other sketches where this does not crash, but all the "intermediate" ribs will have one edge in various states of "twist" and the wire is not closed, so no loft can be made

**Comment by AIRCAP** [comment]
I think this cannot be fixed without changing the way the interpolated shapes are computed - that will likely not be 100% backwards compatible

**Comment by AIRCAP** [comment]
ok, I think this fix (see PR #50 ) should be mostly backwards compatible. there might be borderline cases where previously "Reverse Twist" was needed to correctly apply a shape where now its not, but I don't have any testcases for this. the "Twist" parameter should still work the same as before

**Comment by AIRCAP** [comment]
I think closing this for now makes sense. The example given here is fixed with #50 being merged.
